packages/go_router/lib/src/match.dart

```diff
@@ -61,6 +63,18 @@ class RouteMatch {
     throw MatcherError('Unexpected route type: $route', restLoc);
   }
 
+  /// Completes the promise returned by [GoRouter.push].
+  void complete([T? value]) {
```

Comment:
the entire completer should be move to ImperativeRouteMatch. This can just be a empty implementation for subclass override

Reply:
also maybe change this to didComplete. it is a special keyword in the framework, that has specific meaning

Reply:
Done